benchmark_runner/common/remote_ssh/remote_ssh.py

```diff
@@ -0,0 +1,195 @@
+
+    def mkdir(self, remote_path):
+        """
+        This method mkdir on remote path
+        :param remote_path:
+        :return:
+        """
+        try:
+            self.__p_sftp.chdir(remote_path)  # Test if remote path exists
+        except IOError:
+            self.__p_sftp.mkdir(remote_path)  # Create remote path
+            self.__p_sftp.chdir(remote_path)
+
+    def put_remote_dir(self, local, remote):
+        """
+        This method uploads the contents of the local directory to the remote path. The
+        remote directory needs to exists. All subdirectories in py_perf-code are
+        created under target.
+        :param local: local path
+        :param remote: remote path
+        :return:
+        """
+        for file_name in os.listdir(local):
+            if os.path.isfile(os.path.join(local, file_name)):
+                self.put_remote_file(local, remote, file_name)
+            else:
+                self.mkdir(remote, ignore_existing=True)
+                self.put_remote_file(local, remote, file_name)
+
+    def get_remote_dir(self, remote, local):
+        """
+        This method downloads the contents of the local directory to the remote path. The
+        remote directory needs to exists. All subdirectories in py_perf-code are
+        created under target.
+        :param local: local path
+        :param remote: remote path
+        :return:
+        """
+
+        for file_name in os.listdir(local):
+            if os.path.isfile(os.path.join(local, file_name)):
+                self.get_remote_file(remote, local, file_name)
+            else:
+                self.mkdir(remote, ignore_existing=True)
+                self.get_remote_file(remote, local, file_name)
+
+    def put_remote_file(self, local, remote, file_name):
+        """
+        This method upload file name from local to remote path
+        :param local:
+        :param remote:
+        :param file_name:
+        :return:
+        """
+        self.__p_sftp.put(f'{local}/{file_name}', f'{remote}/{file_name}')
+
+    def get_remote_file(self, remote, local, file_name):
+        """
+        This method download file name from remote to local path
+        :param local:
+        :param remote:
+        :param file_name:
+        :return:
+        """
+        self.__p_sftp.get(f'{remote}/{file_name}', f'{local}/{file_name}')
+
+    def replace_parameter(self, remote_path, file_name, parameter, value, all_line=False):
+        """
+        This method is replace parameter in remote file
+        :param remote_path: path to file
+        :param file_name: file name
+        :param parameter: the parameter
+        :param value: the new value
+        :param all_line: replace all the line that contains the parameter with value
+        :return:
+        """
+        if all_line:
+            res = False
+            for c in ['/', '@', '#', '$', '%', '^', '&', '*', '-', '<', '>', '!', '~', '+', '-']:
```

Comment:
Isn't there some kind of library that will do this safely?

Reply:
What exactly are you trying to do here?

Reply:
Find a special parameter that not parameter, and use it in the replace. otherwise there will be issue in the replace itself.

Reply:
You may want to use shlex.quote:

https://docs.python.org/3/library/shlex.html#shlex.quote

Reply:
I don't understand what you mean by this?

Reply:
If the replace string is a path like home/test. so we cannot use / as separator.  so I replace it by other special character that is not part of replaced parameter.

e.g. sed "s@home/test=.@home/test=8@g" test.txt